src/ops/validate.rs

```diff
@@ -162,11 +162,10 @@ impl ValidateOp for super::Conditional {
         for (i, (child, optype)) in children.into_iter().enumerate() {
             let OpType::Case(case_op) = optype else {panic!("Child check should have already checked valid ops.")};
             let sig = &case_op.signature;
-            let predicate_value = &self.predicate_inputs[i];
-            if sig.input[0..predicate_value.len()] != predicate_value[..]
-                || sig.input[predicate_value.len()..] != self.other_inputs[..]
-                || sig.output != self.outputs
-            {
+            if let Some(expected_inputs) = self.case_input_row(i) {
```

Comment:
Should this be an `unwrap` instead? We know this is a Case op.

Reply:
Hmmm, actually that would work (and solve the problem below too), as the length check above should rule out None

Reply:
Alternatively, I think you could have written
```rust
Some(&sig.input) != self.case_input_row(i).as_ref()
```

Reply:
(Actually the RHS needs to be `(&self.case_input_row(i)).as_ref()` but the key is to put the `as_ref` on the other side, which should have been obvious when I said I needed its inverse...d'oh)